src/server/pfs/server/driver.go

```diff
@@ -2545,3 +2577,9 @@ func (b *branchSet) has(branch *pfs.Branch) bool {
 func has(bs *[]*pfs.Branch, branch *pfs.Branch) bool {
 	return (*branchSet)(bs).has(branch)
 }
+
+// isGlob checks if the pattern contains a glob character
+func isGlob(pattern string) bool {
+	matched, _ := regexp.Match(`[*?\[\]\{\}!]`, []byte(pattern))
```

Comment:
We should compile this Regex once rather than everytime we call `isGlob` it looks like that function will be getting called a lot.